File: intent3/app/src/main/java/com/example/intent3/SecondActivity.java
package com.example.intent3;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        Button b = findViewById(R.id.button2);
        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent();
                EditText e = findViewById(R.id.editText);
                String data = e.getText().toString();
                intent.putExtra("secondactivity", data);
                setResult(Activity.RESULT_OK, intent);
                finish();
            }
        });

    }
}

File: TestKotlin/app/src/main/java/com/example/testkotlin/SecondActivity.kt
package com.example.testkotlin

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import kotlinx.android.synthetic.main.activity_second.*

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val bundle = intent.getBundleExtra("main_activity_data")
        val height = bundle.getFloat("height")
        val weight = bundle.getFloat("weight")

        txt_intentdata.text = "Height: $height | Weight: $weight"

        btn_calculate.setOnClickListener {
            val m_intent = Intent()
            val m_bmi = 703 * (weight / (height * height))
            m_intent.putExtra("second_activity_data", m_bmi)
            setResult(Activity.RESULT_OK, m_intent)
            finish()
        }
    }





}
